darts/models/forecasting/regression_ensemble_model.py

```diff
@@ -149,7 +149,12 @@ def fit(
                 model.fit(**kwargs)
 
             else:
-                model.fit(self.training_series)
+                if (future_covariates is not None) and model.supports_future_covariates:
+                    model.fit(series=series, future_covariates=future_covariates)
+                elif (past_covariates is not None) and model.supports_past_covariates:
+                    model.fit(series=series, past_covariates=past_covariates)
+                else:
+                    model.fit(self.training_series)
```

Comment:
if the `if self.is_global_ensemble` condition is not verified, it indicates that the EnsembleModel contains only `LocalForecastingModels` which do not support past covariates, you can remove this `if-else` case.

Also, you could probably also remove the check on `past_covariate` since a value of `None` will not disrupt the code-flow (as it's the default value in the `fit()` method).

A simpler approach would probably be to reuse the approach for the global_ensemble with something like:
```python
if self.is_global_ensemble:
   kwargs = dict(series=series)
   if model.supports_past_covariates:
      kwargs["past_covariates"] = past_covariates
else:
   kwargs = dict(series=self.training_series)

if model.supports_future_covariates:
   kwargs["future_covariates"] = future_covariates

model.fit(**kwargs)
```

Reply:
Noice!! This is much clean... Will update it and write the test too, Thank you.